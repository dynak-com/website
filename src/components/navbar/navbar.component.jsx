import React from 'react';
import styles from './styles.module.css';
import CustomNavLink from '../navlink/navlink.component';
import { NavLink as Container } from 'react-router-dom';

const Navbar = (props) => {
    const { content, targetId } = props;
    return (
        <nav className={`navbar navbar-expand-lg navbar-dark ${styles.customNavbar}`}>
            <Container to="/">
                <h1 className={`${styles.websiteTitle} navbar-brand`}>
                    dynak<span className={styles.dot}>.</span>com
                </h1>
            </Container>
            <button
                className={`${styles.navbarButton} navbar-toggler`}
                type="button"
                data-toggle="collapse"
                data-target={`#${targetId}`}
                aria-controls={targetId}
                aria-expanded="false"
                aria-label="Toggle navigation"
            >
                <span className="navbar-toggler-icon"></span>
            </button>
            <div className="collapse navbar-collapse" id={targetId}>
                {onRenderNavLinks(content, targetId)}
            </div>
        </nav>
    );
};

export function onRenderNavLinks(data, targetId) {
    return (
        <ul className="navbar-nav ml-auto">
            {data.map((item) => {
                const { text, address } = item;
                return (
                    <li id="item-navlink" className="nav-item" key={text}>
                        <CustomNavLink text={text} address={address} target={targetId} />
                    </li>
                );
            })}
        </ul>
    );
}

export default Navbar;
