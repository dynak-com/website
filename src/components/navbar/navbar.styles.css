#styled-navbar {
  background-color: transparent;
  overflow: hidden;
}

#styled-navbar h1 {
  margin: 0;
  font-size: 2.6rem;
  font-weight: 900;
  font-family: Tenby-five;
}

#styled-navbar span {
    font-size: 1.7rem;
    font-family: Tenby-five;
  }

#navbar-btn {
  border: none;
  outline: none;
}

#item-navlink > a {
  text-decoration: none;
}

.navbar-brand > #dot {
  color: #3eafad;
  font-size: 3rem;
}

.main-container #website-title {
  font-size: 3rem;
}
