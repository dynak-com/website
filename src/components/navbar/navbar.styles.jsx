import React from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";

export const StyledNavbar = styled.nav`
  background-color: transparent;
  overflow: hidden;
  h1 {
    margin: 0;
    font-size: 2.6rem;
    font-weight: 900;
    font-family: Tenby-five;
    cursor: pointer;
    span {
      color: #3eafad;
    }
  }
  span {
    font-size: 1.7rem;
    font-family: Tenby-five;
    cursor: pointer;
  }
`;

const StyledNavLink = styled(NavLink)`
  span {
    transition: 0.3s;
  }
  &:focus,
  &:hover,
  &:visited,
  &:link {
    text-decoration: none;
    &.active > span {
      background-color: #3eafad;
      transform: skew(-12deg);
    }
  }
`;

export default props => <StyledNavLink {...props} />;
