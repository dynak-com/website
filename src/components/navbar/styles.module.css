.customNavbar {
    background-color: transparent;
    overflow: hidden;
}

@media screen and (min-width: 992px) {
    .customNavbar {
        justify-content: flex-start;
    }
}

.customNavbar h1 {
    margin-right: 0;
}

.customNavbar .navbarButton {
    border: none;
    outline: none;
}

.customNavbar span {
    font-size: 1.7rem;
    font-family: Tenby-five;
}

.websiteTitle {
    margin: 0;
    font-size: 2.6rem;
    font-weight: 900;
    font-family: Tenby-five;
}

.websiteTitle .dot {
    color: #3eafad;
    font-size: 2.4rem;
}
