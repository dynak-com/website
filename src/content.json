[
  {
    "lang": "pl",
    "nav": [
      {
        "text": "nav.home",
        "address": "/"
      },
      {
        "text": "nav.about",
        "address": "/about"
      },
      {
        "text": "nav.projects",
        "address": "/projects"
      },
      {
        "text": "nav.contact",
        "address": "/contact"
      }
    ],
    "home": {
      "header": "home.header",
      "posts": [
        {
          "id": 1,
          "text": "home.posts.paragraph1",
          "headerText": "home.posts.headerText1"
        },
        {
          "id": 2,
          "text": "home.posts.paragraph2",
          "headerText": "home.posts.headerText2"
        },
        {
          "id": 3,
          "text": "home.posts.paragraph3",
          "headerText": "home.posts.headerText3"
        }
      ]
    },
    "about": {
      "header": "about.header",
      "technologies": {
        "title": "about.technologies.title",
        "content": "about.technologies.content"
      },
      "team": {
        "title": "about.team.title",
        "teamData": [
          {
            "name": "Emilia Dynak",
            "picture": "https://secure.gravatar.com/avatar/d34a1b21822840695b16113034ba740b?s=180&d=identicon",
            "position": "CEO and co-owner",
            "linkedInUrl": "https://www.linkedin.com/in/emilia-dynak-b0a3a9140/",
            "githubUrl": "https://github.com/edynak",
            "username": "edynak"
          },
          {
            "name": "Cezary Dynak",
            "picture": "https://secure.gravatar.com/avatar/47fbd06257cedcf9594bd9144f948feb?s=180&d=identicon",
            "position": "CTO and co-owner",
            "linkedInUrl": "https://www.linkedin.com/in/cdynak/",
            "githubUrl": "https://github.com/cdynak",
            "username": "cdynak"
          },
          {
            "name": "Natalia Borzyszkowska",
            "picture": "https://secure.gravatar.com/avatar/2cba91020f68596ad56766847694b292?s=180&d=identicon",
            "position": "CFO",
            "linkedInUrl": "https://www.linkedin.com/in/natalia-borzyszkowska-b7814218b/",
            "githubUrl": "https://github.com/login",
            "username": "21_smakow_natki"
          },
          {
            "name": "Aleksandra Borodziuk",
            "picture": "https://secure.gravatar.com/avatar/865d01afdfe703b4ef137693198b8500?s=180&d=identicon",
            "position": "CMO",
            "linkedInUrl": "https://www.linkedin.com/in/aleksandra-borodziuk/",
            "githubUrl": "https://github.com/login",
            "username": "Forestgirl25"
          },
          {
            "name": "Dominika Konefał",
            "picture": "https://secure.gravatar.com/avatar/ec094fb729d8e3509391628d8ba00dd8?s=180&d=identicon",
            "position": "Graphic Designer",
            "linkedInUrl": "https://www.linkedin.com/in/dominika-konefa%C5%82-028920185/",
            "githubUrl": "https://github.com/login",
            "username": "Deloi"
          },
          {
            "name": "Mikołaj Tumalewicz",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/2270247/avatar.png?width=180",
            "position": "DevOps/SysOps Engineer",
            "linkedInUrl": "https://www.linkedin.com/in/mtumalewicz/",
            "githubUrl": "https://github.com/mtumalewicz",
            "username": "mtumalewicz"
          },
          {
            "name": "Marcin Banik",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/2342806/avatar.png?s=180",
            "position": "Front-end Developer",
            "linkedInUrl": "https://www.linkedin.com/in/marcin-banik-947954142/",
            "githubUrl": "https://github.com/m-banik",
            "username": "mbanik"
          },
          {
            "name": "Bartosz Marciniec",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/4684288/avatar.png?s=180",
            "position": "Back-end Developer",
            "linkedInUrl": "https://www.linkedin.com/in/bartosz-marciniec-b0917517a/",
            "githubUrl": "https://github.com/hendas",
            "username": "hendas"
          },
          {
            "name": "Cristina Volk",
            "picture": "https://assets.gitlab-static.net/uploads/-/system/user/avatar/5111565/avatar.png?s=180",
            "position": "Web Developer",
            "linkedInUrl": "https://www.linkedin.com/in/kristina-volk-a9829517b/",
            "githubUrl": "https://github.com/CristinaVolk",
            "username": "CristinaVolk"
          }
        ]
      },
      "values": {
        "title": "about.values.title",
        "content": [
          "about.values.content.paragraph1",
          "about.values.content.paragraph2",
          "about.values.content.paragraph3"
        ]
      }
    },
    "projects": {
      "header": "projects.header",
      "posts": [
        {
          "id": 1,
          "text": "projects.posts.text1",
          "headerText": "dynak.io"
        },
        {
          "id": 2,
          "text": "projects.posts.text2",
          "headerText": "warehouse-app"
        },
        {
          "id": 3,
          "text": "projects.posts.text3",
          "headerText": "projects.posts.headerText3"
        }
      ]
    },
    "contact": {
      "header": "contact.header",
      "socials": "contact.socials",
      "form": {
        "subject": "contact.form.subject",
        "email": "contact.form.email",
        "message": "contact.form.message",
        "contactAgreement": "contact.form.contactAgreement",
        "privacyPolicyAgreement": "contact.form.privacyPolicyAgreement",
        "policyPrivacy": "contact.form.policyPrivacy",
        "send": "contact.form.send"
      }
    },
    "modal": {
      "modal_ok": {
        "modal_id": "modal_ok",
        "error_message": false,
        "thanks": "modal_ok.thanks",
        "sent": "modal_ok.sent",
        "close": "modal_ok.close"
      },
      "modal_error": {
        "modal_id": "modal_error",
        "error_message": true,
        "thanks": "modal_error.thanks",
        "sent": "modal_error.sent",
        "close": "modal_error.close"
      },
      "modal_privacy_policy": {
        "modal_id": "modal_privacy_policy",
        "error_message": null,
        "thanks": "modal_privacy_policy.thanks",
        "sent": "modal_privacy_policy.sent",
        "close": "modal_privacy_policy.close",
        "rodoContent": {
          "header": "rodoContent.header",
          "sentence1": "rodoContent.sentence1",
          "website": "rodoContent.website",
          "sentence2": "rodoContent.sentence2",
          "generalProvisions": "rodoContent.generalProvisions",
          "text1": "rodoContent.text1",
          "text2": "rodoContent.text2",
          "text3": "rodoContent.text3",
          "definitions": "rodoContent.definitions",
          "administrator": "rodoContent.administrator",
          "user": "rodoContent.user",
          "personalData": "rodoContent.personalData",
          "paragraph1": "rodoContent.paragraph1",
          "linkedIn": "rodoContent.linkedIn",
          "paragraph2": "rodoContent.paragraph2",
          "paragraph3": "rodoContent.paragraph3",
          "paragraph4": "rodoContent.paragraph4",
          "paragraph5": "rodoContent.paragraph5",
          "paragraph6": "rodoContent.paragraph6",
          "paragraph7": "rodoContent.paragraph7",
          "paragraph8": "rodoContent.paragraph8",
          "webLinkedIn": "rodoContent.webLinkedIn",
          "github": "rodoContent.github",
          "gitlab": "rodoContent.gitlab",
          "paragraph9": "rodoContent.paragraph9",
          "paragraph10": "rodoContent.paragraph10",
          "paragraph11": "rodoContent.paragraph11",
          "paragraph12": "rodoContent.paragraph12",
          "paragraph13": "rodoContent.paragraph13",
          "paragraph14": "rodoContent.paragraph14",
          "paragraph15": "rodoContent.paragraph15",
          "paragraph16": "rodoContent.paragraph16",
          "paragraph17": "rodoContent.paragraph17",
          "paragraph18": "rodoContent.paragraph18",
          "paragraph19": "rodoContent.paragraph19",
          "paragraph20": "rodoContent.paragraph20",
          "paragraph21": "rodoContent.paragraph21",
          "paragraph22": "rodoContent.paragraph22",
          "paragraph23": "rodoContent.paragraph23",
          "forms": "rodoContent.forms",
          "paragraph24": "rodoContent.paragraph24",
          "paragraph25": "rodoContent.paragraph25",
          "technologies": "rodoContent.technologies",
          "Administrator": "rodoContent.Administrator",
          "not": "rodoContent.not",
          "paragraph26": "rodoContent.paragraph26",
          "cookies": "rodoContent.cookies",
          "part1": "rodoContent.part1",
          "part2": "rodoContent.part2",
          "part3": "rodoContent.part3",
          "part4": "rodoContent.part4",
          "created": "rodoContent.created"
        }
      }
    },
    "error": "error",
    "footer": {
      "rodo": "footer.rodo",
      "dynak": "footer.dynak"
    }
  }
]