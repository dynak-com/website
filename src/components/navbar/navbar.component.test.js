import Navbar from './navbar.component';
import { onRenderNavLinks } from './navbar.component';
import content from '../../content.json';

describe('Navbar  component', () => {
    const [contentData] = content;
    const targetId = 'headerNavbarMenu';

    it('should run onRenderCards function', () => {
        const myMockFn = jest.fn(onRenderNavLinks(contentData.nav, targetId));
        expect(myMockFn).toMatchSnapshot();
    });

    it('renders correctly according to Snapshot', () => {
        const prop = { content: contentData.nav, targetId };
        const myMockFn = jest.fn(Navbar(prop));
        expect(myMockFn).toMatchSnapshot();
    });
});
