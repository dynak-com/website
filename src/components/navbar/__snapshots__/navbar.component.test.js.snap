// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm  component renders correctly according to Snapshot 1`] = `[MockFunction]`;

exports[`ContactForm  component should run onRenderCards function 1`] = `[MockFunction]`;

exports[`Navbar  component renders correctly according to Snapshot 1`] = `[MockFunction]`;

exports[`Navbar  component should run onRenderCards function 1`] = `[MockFunction]`;
